statsmodels/base/data.py

```diff
@@ -367,7 +367,11 @@ def _get_row_labels(self, arr):
 
     def _get_names(self, arr):
         if isinstance(arr, DataFrame):
-            return list(arr.columns)
+            if isinstance(arr.columns, MultiIndex):
+                # Flatten MultiIndexes into "simple" column names
+                return [".".join((level for level in c if level)) for c in arr.columns]
```

Comment:
wouldn't underline instead of a dot be better as joining character?
I usually use underline to keep it as valid python name (if it already is), but I don't know of any conventions for this.

Reply:
Pandas uses underscore for these scenarios. Like Groupby('z').describe() would create 'x_mean' or when you merge/join it can create x_left and x_right for example.